opencv/Dockerfile

```diff
@@ -1,69 +1,59 @@
-FROM hudl/ffmpeg
-
+FROM hudl/ffmpeg:latest
 MAINTAINER Ryan Versaw <ryan.versaw@hudl.com>
 
-# Dependencies
 RUN apt-get update && apt-get install -y \
     cmake \
     gfortran \
     libatlas-base-dev \
-    python \
-    python-dev \
     wget
 
-# Download OpenCV and extra modules
+RUN apt-get install software-properties-common -y && \
+    add-apt-repository ppa:george-edison55/cmake-3.x && \
+    apt-get update && \
+    apt-get upgrade cmake -y
+
+RUN wget --quiet https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O install_miniconda.sh && \
+    /bin/bash install_miniconda.sh -b -p /opt/conda && \
+    rm install_miniconda.sh
+ENV PATH /opt/conda/bin:$PATH
+
+RUN pip install --upgrade pip
+RUN pip install Cython==0.24
+RUN pip install --upgrade --ignore-installed https://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-0.11.0-cp35-cp35m-linux_x86_64.whl
+
+RUN conda install numpy -y
+
 WORKDIR /
 RUN git clone https://github.com/Itseez/opencv.git
 WORKDIR /opencv
-RUN git checkout tags/3.1.0
+RUN git checkout tags/3.2.0
 
 WORKDIR /
 RUN git clone https://github.com/Itseez/opencv_contrib.git
 WORKDIR /opencv_contrib
-RUN git checkout tags/3.1.0
+RUN git checkout tags/3.2.0
 
-# Build OpenCV
 WORKDIR /opencv
-RUN cmake -DOPENCV_EXTRA_MODULES_PATH=../opencv_contrib/modules . && \
-    make && make install && \
+RUN cmake -D CMAKE_BUILD_TYPE=Release \
```

Comment:
I have a few other options in my local build:
```
-D WITH_CUDA=ON
-D ENABLE_FAST_MATH=1
-D CUDA_FAST_MATH=1
-D WITH_CUBLAS=1
```
Are those ones we should look at adding? I'm not actually sure what the defaults on those options are, so I don't know if it makes much of a difference.